File: src/contests/cg/puzzle/python3/rubik.py
# 🎮 CodinGame Puzzle - rubik
# https://www.codingame.com/training/medium/rubik

n = int(input())
# Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
inner = n - 2 if n >= 2 else 0
print(n ** 3 - inner ** 3)

File: src/contests/cg/puzzle/lua/rubik.lua
-- 🎮 CodinGame Puzzle - rubik
-- https://www.codingame.com/training/medium/rubik

local n = tonumber(io.read())
-- Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
local inner = n >= 2 and n - 2 or 0
print(n * n * n - inner * inner * inner)
